packages/block-library/src/query/edit/inspector-controls/order-control.js

```diff
@@ -4,27 +4,37 @@
 import { SelectControl } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
-const orderOptions = [
-	{
-		label: __( 'Newest to oldest' ),
-		value: 'date/desc',
-	},
-	{
-		label: __( 'Oldest to newest' ),
-		value: 'date/asc',
-	},
-	{
-		/* translators: label for ordering posts by title in ascending order */
-		label: __( 'A → Z' ),
-		value: 'title/asc',
-	},
-	{
-		/* translators: label for ordering posts by title in descending order */
-		label: __( 'Z → A' ),
-		value: 'title/desc',
-	},
-];
-function OrderControl( { order, orderBy, onChange } ) {
+function OrderControl( { order, orderBy, postType, onChange } ) {
+	const orderOptions = [
+		{
+			label: __( 'Newest to oldest' ),
+			value: 'date/desc',
+		},
+		{
+			label: __( 'Oldest to newest' ),
+			value: 'date/asc',
+		},
+		{
+			/* translators: label for ordering posts by title in ascending order */
+			label: __( 'A → Z' ),
+			value: 'title/asc',
+		},
+		{
+			/* translators: label for ordering posts by title in descending order */
+			label: __( 'Z → A' ),
+			value: 'title/desc',
+		},
+		// Only include the menu_order option for pages
+		...( postType === 'page'
```

Comment:
This would also need to be adjusted to check for `page-attributes` flag of the post type instead.